src/Interpreters/Context.cpp

```diff
@@ -1446,6 +1446,31 @@ void Context::addWarningMessageAboutDatabaseOrdinary(const String & database_nam
         shared->addWarningMessage(message);
 }
 
+void Context::setRemoteReadThrottler(size_t bandwidth)
+{
+    setThrottler(shared->remote_read_throttler, bandwidth);
+}
+
+void Context::setRemoteWriteThrottler(size_t bandwidth)
+{
+    setThrottler(shared->remote_write_throttler, bandwidth);
+}
+
+void Context::setThrottler(std::shared_ptr<Throttler> & throttler, size_t bandwidth)
+{
+    std::lock_guard lock(shared->mutex);
+
+    if (bandwidth > 0)
+    {
+        if (!throttler)
+            throttler = std::make_shared<Throttler>(bandwidth);
+        else
+            throttler->setMaxSpeed(bandwidth);
+    }
+    else
+        throttler.reset();
```

Comment:
I think this is incorrect, if bandwidth = 0, it should mean infinity. That said, you still need to create the throttler itself, because there are many places where it is used without checking for nullptr. 

Check out getRemoteReadThrottler()

Reply:
afaiu, if bandwidth = 0, the throttler is nil now. but indeed the code has some problem here.
i am going to try to reuse getRemoteReadThrottler